backend/src/Designer/Controllers/DatamodelsController.cs

```diff
@@ -207,8 +208,13 @@ public async Task<IActionResult> UseXsdFromRepo(string org, string repository, s
             Guard.AssertFileExtensionIsOfType(filePath, ".xsd");
 
             string xsd = await _schemaModelService.GetSchema(org, repository, developer, filePath);
-            var xsdStream = new MemoryStream(Encoding.UTF8.GetBytes(xsd ?? string.Empty));
-            string jsonSchema = await _schemaModelService.BuildSchemaFromXsd(org, repository, developer, filePath, xsdStream);
+            if (xsd == null)
```

Comment:
Here the custom exception can be handled instead of handling it in service and returning null.